artiq/dashboard/datasets.py

```diff
@@ -81,6 +82,89 @@ def get_edit_widget(self, initial_value):
     def get_edit_widget_value(self):
         return self.edit_widget.text()
 
+class Creator(QtWidgets.QDialog):
+    def __init__(self, parent, dataset_ctl):
+        QtWidgets.QDialog.__init__(self, parent=parent)
+        self.dataset_ctl = dataset_ctl
+
+        self.setWindowTitle("Create dataset")
+        grid = QtWidgets.QGridLayout()
+        grid.setRowMinimumHeight(1, 35)
+        grid.setColumnMinimumWidth(2, 60)
+        self.setLayout(grid)
+
+        grid.addWidget(QtWidgets.QLabel("Name:"), 0, 0)
+        grid.addWidget(self.key_widget(), 0, 1)
+
+        grid.addWidget(QtWidgets.QLabel("Value:"), 1, 0)
+        grid.addWidget(self.edit_widget(), 1, 1)
+        self.data_type = QtWidgets.QLabel("data type")
+        grid.addWidget(self.data_type, 1, 2)
+        self.value_widget.textChanged.connect(self.dtype)
+
+        grid.addWidget(QtWidgets.QLabel("Persist:"), 2, 0)
+        grid.addWidget(self.persist_widget(), 2, 1)
+
+        self.ok = QtWidgets.QPushButton('&Ok')
+        self.ok.setEnabled(False)
+        self.cancel = QtWidgets.QPushButton('&Cancel')
+        self.buttons = QtWidgets.QDialogButtonBox(self)
+        self.buttons.addButton(self.ok, QtWidgets.QDialogButtonBox.AcceptRole)
+        self.buttons.addButton(self.cancel, QtWidgets.QDialogButtonBox.RejectRole)
+        grid.setRowStretch(3, 1)
+        grid.addWidget(self.buttons, 4, 0, 1, 3)
+        self.buttons.accepted.connect(self.accept)
+        self.buttons.rejected.connect(self.reject)
+
+    def accept(self):
+        try:
+            key = self.get_create_widget_text()
+            value = self.get_create_widget_value()
+            persist = self.get_persist_widget_value()
+            asyncio.ensure_future(self.dataset_ctl.set(key, pyon.decode(value), persist))
```

Comment:
Use this to wrap the asyncio task and log errors immediately: https://github.com/m-labs/artiq/blob/master/artiq/tools.py#L107
Otherwise the asyncio task exceptions are only retrieved when the GUI exits and error messages printed then. This makes debugging tedious.